snuba/migrations/operations.py

```diff
@@ -9,26 +10,48 @@
 from snuba.migrations.table_engines import TableEngine
 
 
+class OperationTarget(Enum):
+    """
+    Represents the target nodes of an operation.
+    Either local, distributed.
+    """
+
+    LOCAL = "local"
+    DISTRIBUTED = "distributed"
+
+
 class SqlOperation(ABC):
     def __init__(
-        self, storage_set: StorageSetKey, settings: Optional[Mapping[str, Any]] = None
+        self,
+        storage_set: StorageSetKey,
+        settings: Optional[Mapping[str, Any]] = None,
+        target: OperationTarget = OperationTarget.LOCAL,
```

Comment:
I think this is a bit dangerous, the user should explicitly provide the target. Can we do away with the default?

Reply:
I think all the child classes of SqlOperation need take the target argument in the constructor now. I think we should try and keep this immutable.

Reply:
The trouble with having no default is we would have to add the parameter in all 110 of the migrations files for the existing migrations. What I am doing instead is setting a default the applying a wrapper to correct the target for the legacy migrations.

Reply:
The fact that not all child classes take this argument means it would still be wrong for any new migrations though.

I know it's a bit of grunt work but one option would be to have a separate PR that gets merged first that only adds the OperationTarget enum and makes it mandatory in any SqlOperation. I know this is annoying but some reasons for proposing this:
- I am not a fan of making things that should be immutable mutable and would like to find a way to avoid it
- I think having local as a default is a bit of a footgun and would actually prefer making the default None and then adding some validation later to ensure it is set even if we absolutely need to keep it mutable
- If not now I think we would just need to do this work later to make it a required argument as it affects the interface of the SqlOperation class going forward

Reply:
Having target as a required is super tricky cause I can't see a way around manually adding it to every migration. Having default None is more doable and would throw errors on new migrations if it's not explicitly set. I think thats the better way to go for now.

Reply:
For typing purposes, setting None as default can be done by adding an UNSET enum to the OperationTarget. That also makes it more clear that it needs to be set.